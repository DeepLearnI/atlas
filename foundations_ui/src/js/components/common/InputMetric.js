import React, { Component } from 'react';
import PropTypes from 'prop-types';
import TableSectionHeader from './TableSectionHeader';
import CommonActions from '../../actions/CommonActions';

class InputMetric extends Component {
  constructor(props) {
    super(props);
    this.state = {
      header: this.props.header,
      hiddenInputParams: this.props.hiddenInputParams,
      allInputParams: this.props.allInputParams,
    };
  }

  render() {
    const { header, hiddenInputParams, allInputParams } = this.state;

    const inputParams = CommonActions.getInputMetricColumnHeaders(allInputParams);

    return (
      <div className="input-metric-container">
        <TableSectionHeader header={header} />
        <div className="input-metric-column-header-container">
          {inputParams}
        </div>
      </div>
    );
  }
}

InputMetric.propTypes = {
  header: PropTypes.string,
  hiddenInputParams: PropTypes.array,
  allInputParams: PropTypes.array,
};

InputMetric.defaultProps = {
  header: '',
  hiddenInputParams: [],
  allInputParams: [],
};


export default InputMetric;
