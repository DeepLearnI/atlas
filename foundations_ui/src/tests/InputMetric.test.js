import React from 'react';
import ReactDOM from 'react-dom';
import InputMetric from '../js/components/common/InputMetric';
import { shallow, mount } from 'enzyme';
import configureTests from './setupTests';

configureTests();

it('Shallow Renders InputMetric', () => {
  const wrapper = shallow(<InputMetric/>);
  expect(wrapper).toMatchSnapshot();
});