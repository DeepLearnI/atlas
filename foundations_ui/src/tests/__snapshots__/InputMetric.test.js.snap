// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow Renders InputMetric 1`] = `
<div
  className="input-metric-container"
>
  <TableSectionHeader
    header=""
  />
  <div
    className="input-metric-column-header-container"
  />
</div>
`;
